nbody_gravitational/NBodyGravitational.jl

```diff
@@ -0,0 +1,173 @@
+module NBodyGravitational
+
+using DiffEqBase, DifferentialEquations, StaticArrays, Plots, StaticArrays
+
+
+grav_const = 6.673e-11
+
+#=
+    Represents a body/particle in an N-body gravitational problem
+    m - mass
+    r - initial position
+    v - initial velocity
+=#
+struct MassBody{mType, cType}
+    m :: mType
+    r :: Vector{cType}
+    v :: Vector{cType}
+    
+    function MassBody{mType,cType}(m::mType, r::Vector{cType}, v::Vector{cType})
+        @assert length(r)==length(v) "The length of the position (r) and velocity (v) components should match"
+        new(m,r,v)
+    end
+end
+
+MassBody(m::mType, r :: Vector{cType}, v :: Vector{cType}) where {mType <: AbstractFloat, cType  <: AbstractFloat} = MassBody{mType, cType}(m,r,v)
+
+#=
+    Contains the necessary information for solving an N-body gravitational problem
+    bodies - interacting bodies/particles
+    tSpan - time, for which we want to evaluate the solution
+    G - gravitational constant
+=#
+struct NBodyGravProblem
+    bodies :: Array{MassBody, 1}
+    tSpan :: Tuple{Float64, Float64}
+    G :: Float64
+end
+
+NBodyGravProblem(bodies, tSpan::Tuple{Float64, Float64}) = NBodyGravProblem(bodies, tSpan, grav_const)
+
+#= 
+    As a quick approach to solving an N-body problem we will just 
+    transform NBodyGravProblem to an ODEProblem
+
+    Here u will be an array consisting of a consecutive positional coordinates r 
+    following by components of velocities v
+=#
+import Base.convert
+convert(::Type{ODEProblem}, x:: NBodyGravProblem) =
+begin
```

Comment:
It probably makes sense to define the constructor for a SecondOrderODEProblem

Reply:
I agree. I have added it.